app/src/main/java/ch/epfl/culturequest/database/Database.java

```diff
@@ -549,4 +551,76 @@ public static CompletableFuture<Follows> getFollowed(String UId) {
         });
         return future;
     }
+
+    /**
+     * This method is used to retrieve an artwork when scanning. It is intentionally not completing
+     * the future if the artwork is not found, so that the api fetch can succeed before this one fails.
+     *
+     * @param artName the name of the artwork to be retrieved
+     * @return a CompletableFuture that will be completed when the artwork is retrieved
+     */
+    public static CompletableFuture<BasicArtDescription> getArtworkScan(String artName) {
```

Comment:
We do not wait if both fails, the only problem I had is that if firebase fails, the way acceptEither is built makes the future stop, though we can still wait for the api. so for scanning, I made this particular method to never fail, so that acceptEither waits for the other result and not stop.